chapter6/elo-leaderboard/test_load_battles_empty.py

```
@@ -0,0 +1,34 @@
+    path.write_text("[]", encoding="utf-8")
+    df = cli._load_battles(str(path))
+    assert list(df.columns) == ["model_a", "model_b", "winner"]
+    assert len(df) == 0
+
+
+def test_load_battles_nonempty_still_requires_columns(tmp_path):
+    path = tmp_path / "bad.json"
+    path.write_text(json.dumps([{"x": 1}]), encoding="utf-8")
+    try:
+        cli._load_battles(str(path))
+        assert False, "expected ValueError"
```

Ruff (0.15.21)

[warning] Do not `assert False` (`python -O` removes these calls), raise `AssertionError()`

Replace `assert False`

(B011)